### PR TITLE
Caching: Add feature toggle for memory efficient cache payload serialization

### PR DESCRIPTION
**Pull request by mmandrus**
**What is this feature?**

Creates a new feature toggle used by the enterprise caching feature.

### CODE REVIEW

**Review by jcalisto — APPROVED** [review]

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 1]
```diff
@@ -19,38 +19,39 @@ This page contains a list of available feature toggles. To learn how to turn on
...
+| `redshiftAsyncQueryDataSupport`                  | Enable async query data support for Redshift                                                                                                                                                                                                                                      | Yes                |
+| `athenaAsyncQueryDataSupport`                    | Enable async query data support for Athena                                                                                                                                                                                                                                        | Yes                |
+| `cloudwatchNewRegionsHandler`                    | Refactor of /regions endpoint, no user-facing changes                                                                                                                                                                                                                             | Yes                |
```
Comment (knylander-grafana) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:30]:
```suggestion
| `cloudwatchNewRegionsHandler`                    | Refactor of `/regions` endpoint, no user-facing changes                                                                                                                                                                                                                             | Yes                |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 2]
```diff
@@ -19,38 +19,39 @@ This page contains a list of available feature toggles. To learn how to turn on
...
+| `nestedFolderPicker`                             | Enables the new folder picker to work with nested folders. Requires the nestedFolders feature flag                                                                                                                                                                                | Yes                |
+| `accessTokenExpirationCheck`                     | Enable OAuth access_token expiration check and token refresh using the refresh_token                                                                                                                                                                                              |                    |
```
Comment (knylander-grafana) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:32]:
```suggestion
| `cloudwatchNewRegionsHandler`                    | Refactor of `/regions` endpoint, no user-facing changes                                                                                                                                                                                                                             | Yes                |
| `nestedFolderPicker`                             | Enables the new folder picker to work with nested folders. Requires the `nestedFolders` feature toggle                                                                                                                                                                                | Yes                |
| `accessTokenExpirationCheck`                     | Enable OAuth access_token expiration check and token refresh using the `refresh_token`                                                                                                                                                                                              |                    |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 3]
```diff
@@ -19,38 +19,39 @@ This page contains a list of available feature toggles. To learn how to turn on
...
+| `prometheusMetricEncyclopedia`                   | Adds the metrics explorer component to the Prometheus query builder as an option in metric select                                                                                                                                                                                 | Yes                |
+| `prometheusResourceBrowserCache`                 | Displays browser caching options in Prometheus data source configuration                                                                                                                                                                                                          | Yes                |
+| `prometheusDataplane`                            | Changes responses to from Prometheus to be compliant with the dataplane specification. In particular it sets the numeric Field.Name from 'Value' to the value of the `__name__` label when present.                                                                               | Yes                |
```
Comment (knylander-grafana) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:39]:
```suggestion
| `prometheusDataplane`                            | Changes responses to from Prometheus to be compliant with the dataplane specification. In particular it sets the `numeric Field.Name` from `Value` to the value of the `__name__` label when present.                                                                               | Yes                |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 4]
```diff
@@ -19,38 +19,39 @@ This page contains a list of available feature toggles. To learn how to turn on
...
+| `dataplaneFrontendFallback`                      | Support dataplane contract field name change for transformations and field name matchers where the name is different                                                                                                                                                              | Yes                |
+| `alertingNotificationsPoliciesMatchingInstances` | Enables the preview of matching instances for notification policies                                                                                                                                                                                                               | Yes                |
+| `useCachingService`                              | When turned on, the new query and resource caching implementation using a wire service inject will be used in place of the previous middleware implementation                                                                                                                     |                    |
```
Comment (knylander-grafana) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:43]:
```suggestion
| `useCachingService`                              | When active, the new query and resource caching implementation using a wire service inject replaces the previous middleware implementation.                                                                                                                     |                    |
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md (hunk 3 above)
Comment (knylander-grafana) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:39]:
```suggestion
| `prometheusDataplane`                            | Changes responses to from Prometheus to be compliant with the dataplane specification. In particular, when this feature toggle is active, the numeric `Field.Name` is set from 'Value' to the value of the `__name__` label.                                                                               | Yes                |
```

**Review by knylander-grafana — COMMENTED** [review]
Approving the doc change with some minor suggestions. Thank you for updating the feature toggle description and transferring these suggested changes to the registry.go file so they can be included when the page is next autogenerated.

**Pull request merged**